### PR TITLE
Add plugin option for global filters

### PR DESCRIPTION
**Pull request by cchantep**
Support minimal global filter as plugin option `-P:silencer:globalFilters=[semi-colon separated message patterns]`

See [example](https://github.com/cchantep/silencer/tree/cch/global-filters-example)

### CODE REVIEW

File: silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-import scala.collection.mutable.ArrayBuffer
+import scala.util.matching.Regex
+
+import scala.collection.mutable, mutable.ArrayBuffer
+
 import scala.reflect.internal.util.{Position, SourceFile}
 import scala.tools.nsc.reporters.Reporter
 
-class SuppressingReporter(original: Reporter) extends Reporter {
+class SuppressingReporter(
```
[the remark this answers is not in the document]
Comment (cchantep) [reply silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala]: Such line would exceed 80cols

File: silencer-plugin/testdata/globallyFiltered.scala [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package testdata
+
+object globallyFiltered {
```
[the remark this answers is not in the document]
Comment (cchantep) [reply silencer-plugin/testdata/globallyFiltered.scala]: Updated

File: silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghik) [reply silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala]: I don't have such limit in `silencer` project, currently I have formatting enforced by IntelliJ and it limits lines to 120 columns which for me is more sensible. You can also move `extends Reporter {` to next line but leave parameters in a single line.

**Comment by ghik** [comment]
Good job, thanks a lot!

**Comment by SethTisue** [comment]
looks useful — would like to try it out in a release

**Comment by SethTisue** [comment]
should this PR have included README changes?

**Comment by ghik** [comment]
I'll try to update README and do a release until the end of this week.

**Pull request merged**